File: backend/routes/report.py
from flask import Blueprint, request, jsonify
from db import get_db_connection

report_bp = Blueprint("report", __name__)

@report_bp.route("/sales/report", methods=["GET"])
def sales_report():
    username = request.args.get("username")

    conn = get_db_connection()
    cursor = conn.cursor(dictionary=True)

    # 获取用户ID
    cursor.execute("SELECT user_id FROM users WHERE username = %s", (username,))
    user = cursor.fetchone()
    if not user:
        return jsonify({"message": "User not found"}), 404
    user_id = user["user_id"]

    # 查询销售记录（关联药品名）
    cursor.execute("""
        SELECT 
            s.drug_id,
            d.name AS drug_name,
            d.manufacturer AS drug_manufacturer,
            s.quantity,
            d.price,
            (d.price * s.quantity) AS total_price, 
            s.sale_time
        FROM sales s
        JOIN drugs d ON s.drug_id = d.drug_id
        WHERE s.seller_id = %s
        ORDER BY s.sale_time DESC
    """, (user_id,))
    sales = cursor.fetchall()

    # 计算总销售额
    total = sum(s["quantity"] * float(s["price"]) for s in sales)

    conn.close()

    return jsonify({
        "sales": sales,
        "total_sales": round(total, 2)
    })

File: fronted-login/js/report.js
document.addEventListener("DOMContentLoaded", () => {
  const username = sessionStorage.getItem("username");
  if (!username) {
    document.body.innerHTML = "<h2>请先登录</h2>";
    return;
  }

  console.log("当前用户名:", username);
  const reportTable = document.getElementById("reportTable");
  const totalDisplay = document.getElementById("total");

  fetch(`http://127.0.0.1:5000/sales/report?username=${username}`)
    .then(res => res.json())
    .then(data => {
      if (!data.sales || data.sales.length === 0) {
        totalDisplay.innerText = "暂无销售记录。";
        return;
      }

      const rows = data.sales.map(sale => `
        <tr>
          <td>${sale.drug_id}</td>
          <td>${sale.drug_name}</td>
          <td>${sale.drug_manufacturer}</td>
          <td>${sale.quantity}</td>
          <td>${sale.price}</td>
          <td>¥${sale.total_price}</td>
          <td>${sale.sale_time}</td>
        </tr>
      `).join("");
      reportTable.innerHTML = rows;
      totalDisplay.innerText = `总销售额：￥${data.total_sales}`;
    })
    .catch(err => {
      console.error("加载销售报告失败:", err);
      totalDisplay.innerText = "加载失败，请检查网络或服务器。";
    });
});
